src/parseline.c

```diff
@@ -838,27 +839,15 @@ void m_nick (char *sender, char *tail)
     uptr2 = find_user(newnick);
     if (uptr2) {
         if (uptr && (find_link2(oldnick, newnick) || find_link2(newnick, oldnick)) && uptr->authed == 1) {
```

Comment:
Can you add a comment stating the intended behavior of this scenario (an authenticated user changing nick towards a linked nick)? Also, what is the intended meaning of the umode +r from now on?